cli/src/cli_util.rs

```diff
@@ -1200,6 +1200,24 @@ impl WorkspaceCommandHelper {
         template.format(commit, formatter)
     }
 
+    /// Writes one-line summary of the given `commit`.
+    #[instrument(skip_all)]
+    pub fn write_commit_summary_no_branches(
+        &self,
+        formatter: &mut dyn Formatter,
+        commit: &Commit,
+    ) -> std::io::Result<()> {
+        let template = parse_commit_summary_template_no_branches(
+            self.repo().as_ref(),
+            self.workspace_id(),
+            self.id_prefix_context(),
+            &self.template_aliases_map,
+            &self.settings,
+        )
+        .expect("parse error should be confined by WorkspaceCommandHelper::new()");
+        template.format(commit, formatter)
```

Comment:
Since this template is specific to `jj branch list` command, it's probably better to parse template in `cmd_branch_list()` and handle error. See `workspace_command.parse_commit_template()`.

Reply:
I did this rather mechanically, let me know if it's not what you had in mind.

Reply:
You can inline `write_commit_summary_no_branches()` into `cmd_branch_list()`, do parsing only once, and call `template.format()` in a loop.

Reply:
Good point! Done.